### DISCUSSION TITLE
Fixed-size Ui with fixed-size Frame?

### DISCUSSION THREAD

**Discussion by johan-overbye** [emilk/egui#2315] [Q&A]
I'm struggling to create a fixed-size `Ui` with a fixed-size `Frame`. What I'd like to do is given an `Ui` A, I'd like to create a `Ui` B inside of A, at a location and size of my choice, whose bounds does not grow under any circumstances due to layout needs.

What I'm trying at the moment, inspired by [this discussion](https://github.com/emilk/egui/discussions/469#discussioncomment-836880), is to create `Ui` B using the [`child_ui`](https://docs.rs/egui/latest/egui/struct.Ui.html#method.child_ui) method, then call `Frame::canvas(&Style::default()).show(&mut ui_b, |ui| { ui.vertical(|ui| {/*...*/}) })`.

The frame looks right to begin with, however if I lay out more widgets than will fit vertically in the `Frame`, the `Frame` grows. The same thing happens if I don't create a `Frame`, but add the widgets directly to `ui_b`.

How can I make the `Frame` stay a fixed size?

---

I imagine I could solve it by using [`Ui::put`](https://docs.rs/egui/latest/egui/struct.Ui.html#method.put) to manually place each widget. But that's more fiddly and verbose than I'd like. Ideally I'd like to say "this Ui needs to be this exact size and at this exact position in its parent Ui, but feel free to lay out Widgets inside the Ui as per the normal egui rules".

---

Turns out that forcing widget positions with `Ui::put` isn't quite viable for me: I need to place a `ComboBox` in the frame, and `ComboBox` is not a `Widget`.

**Comment by yunfengwangluo** [comment]
ui.set_min_size(vec2(600.0, 600.0));
ui.set_max_size(vec2(600.0, 600.0));

**Reply by johan-overbye** [reply]
I've tried `set_max_size` in several different `Ui`s, but have not been able to constrain the `Ui` size. It appears to make no difference. I realise a code example would be useful here, but I don't have time to put one together at the moment. Also I'm leaning away from egui for my project at this point, as the layout rules are confusing and seemingly inconsistent.

**Comment by xlambein** [comment]
I'm struggling with the exact same thing. I have layouting constraints, where I want to prevent some elements (e.g., buttons, combo-boxes) from using more space than is given to them, but I can't seem to find a straightforward way to do this. Many of the methods that seem to constrain the layout (e.g. `set_max_size` referenced above) don't actually seem to do anything 😕
